CMakeLists.txt

```diff
@@ -251,7 +251,12 @@ if(ENABLE_TRACY)
 endif()
 
 add_library(metal_header_directories INTERFACE)
-target_include_directories(metal_header_directories INTERFACE ${PROJECT_SOURCE_DIR}/tt_metal/hw/inc)
+target_include_directories(
+    metal_header_directories
+    INTERFACE
+        ${PROJECT_SOURCE_DIR}/tt_metal/hw/inc
+        ${PROJECT_SOURCE_DIR}/tt_metal/include
```

Comment:
Personal preference FWIW: I find `api` more clear/obvious.

This is based on past experience where some devs came from a background of `inc/` and `src/` and would toss ALL headers in `inc/` "because that's where headers go", and public API and internal includes would get mixed up, the public API would leak internals, and it was generally a mess.  A simple rename to `api/` and `src/` had a dramatic reduction in confusion and discussions and newcomers were no longer confused about the API.

Bonus: 'api' sorts higher than (most) other directories, giving it a more prominent placement in the directory tree.

Reply:
Noted. We had internal discussion on this topic, and my understanding was that we had settled on `include` rather than `api` (which was one of the options considered in that discussion). I understand the points you're making. If you feel strongly about this, we should pick up this discussion again, and should deal with that decision orthogonally to this PR, since I do not intend to move files around as part of this code change.